modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/EndpointDefinitionFactory.java

```diff
@@ -267,26 +276,32 @@ public EndpointDefinition createDefinition(OMElement elem) {
                 SynapseConstants.SYNAPSE_NAMESPACE,
                 XMLConfigConstants.ERROR_CODES));
             if (suspendCodes != null && suspendCodes.getText() != null) {
-
-                StringTokenizer st = new StringTokenizer(suspendCodes.getText().trim(), ", ");
-                while (st.hasMoreTokens()) {
-                    String s = st.nextToken();
-                    try {
-                        definition.addSuspendErrorCode(Integer.parseInt(s));
-                    } catch (NumberFormatException e) {
-                        handleException("The suspend error codes should be specified " +
-                            "as valid numbers separated by commas : " + suspendCodes.getText(), e);
+                Value suspendErrorCodesValue = valueFactory.createTextValue(suspendCodes);
+                if (suspendErrorCodesValue.getKeyValue() != null) {
+                    StringTokenizer st = new StringTokenizer(suspendCodes.getText().trim(), ",");
+                    while (st.hasMoreTokens()) {
+                        String s = st.nextToken();
+                        try {
+                            Integer.parseInt(s);
+                        } catch (NumberFormatException e) {
+                            handleException("The suspend error codes should be specified " +
+                                    "as valid numbers separated by commas : " + suspendCodes.getText(), e);
+                        }
                     }
                 }
+                definition.setSuspendErrorCodes(valueFactory.createTextValue(suspendCodes));
```

Comment:
We should be able to resuse the variable in line 279

Reply:
Updated.